### PR TITLE
Fix inverted condition for min_tile_cols.

### LINKED ISSUE #1447: --tile-cols-log2 not working as expected
I'm running `./target/release/rav1`, built with the latest code. When I use `--tiles 4`, the encoding has two rows and two columns, as expected. But when I use ` --tile-cols-log2 1 --tile-rows-log2 1`, the encoding has two rows but only one column.

I expect it to be the same as `--tiles 4`. Is ` --tile-cols-log2 1 --tile-rows-log2 1` the right way to specify two rows and two columns?

### PR DESCRIPTION
**Pull request by tdaede**
Fixes #1447

### CODE REVIEW

File: src/encoder.rs [hunk 1]
```diff
@@ -524,7 +524,7 @@ impl<T: Pixel> FrameInvariants<T> {
       sequence.sb_size_log2(),
       config.width,
       config.height,
-      config.tile_cols_log2.min(min_tile_cols),
+      config.tile_cols_log2.max(min_tile_cols),
```
Comment (rom1v) [inline src/encoder.rs]: IMO, the caller should only provide `config.tile_cols_log2`, it's the resposability of `TilingInfo` internals to clip the requested value if necessary (it already does that for other constraints).

Moreover, it seems wrong to min (or max) a number of cols in log2 with a number of cols (not in log2).

Comment (tdaede) [reply src/encoder.rs]: yeah it looks like that part of the condition was always wrong. I'll move it inside TilingInfo and fix it.

Comment (tdaede) [reply src/encoder.rs]: In fact you already wrote the exact same condition in TilingInfo, so I just deleted this one entirely.

**Comment by tdaede** [comment]
rebased

**Pull request merged**